src/configure/configure.ts

```diff
@@ -443,16 +443,41 @@ class Orchestrator {
         //Post selecting the template update this.inputs.repositoryAnalysisApplicationSettings with corresponding languageSettings
         if (!!repoAnalysisResult
             && !!repoAnalysisResult.repositoryAnalysisApplicationSettingsList) {
-
+            
             //Get languageSettings (corresponding to language of selected settings) provided by RepoAnalysis
-            this.inputs.repositoryAnalysisApplicationSettings = repoAnalysisResult.repositoryAnalysisApplicationSettingsList.find(applicationSettings => {
-                return applicationSettings.language === this.inputs.pipelineConfiguration.template.language;
-            });
+            this.updateRepositoryAnalysisApplicationSettings(repoAnalysisResult);
         }
 
         telemetryHelper.setTelemetry(TelemetryKeys.ChosenTemplate, this.inputs.pipelineConfiguration.template.label);
     }
 
+    private async updateRepositoryAnalysisApplicationSettings(repoAnalysisResult: RepositoryAnalysisParameters): Promise<void>{
+        var applicationSettings = repoAnalysisResult.repositoryAnalysisApplicationSettingsList.filter(applicationSetting => {
+            return applicationSetting.language === this.inputs.pipelineConfiguration.template.language;
+        });
+        
+        let workspacePaths = Array.from(new Set(applicationSettings.map(a => a.settings.workingDirectory)));
```

Comment:
`applicationSettings.map(a => a.settings.workingDirectory)` should itself return an array. Why do you need to do `Array.from(new Set(...))` ?

Reply:
We don't want same entry more than once in our array. Hence we are first converting string to HashSet and then converting it back to array.